### PR TITLE
Adds API reference

### PR DESCRIPTION
**Pull request by abehaskins**
Moves the Angularfire API reference into this repository

### CODE REVIEW

**Comment by jwngr** [comment]
Looks good to me Abe! Back to you for merging. Please use a squash commit :)

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -93,7 +93,7 @@ $ grunt install             # install Selenium server for end-to-end tests
 $ export ANGULARFIRE_TEST_DB_URL="https://<YOUR-DATABASE-NAME>.firebaseio.com"
 ```
 
-3. Update the entire `config` variable in [`tests/initialize.js`](../tests/initialize.js) to
+3. Update the entire `config` variable in [`tests/initialize.js`](/tests/initialize.js) to
```
Comment (katowulf) [inline .github/CONTRIBUTING.md]: This link can't be right? Should be `tests/initialize.js` in both places?

File: docs/guide/beyond-angularfire.md [hunk 2]
```diff
@@ -77,10 +72,10 @@ or you can host it at any custom domain on one of our paid plans. Check out
 
 There are many additional resources for learning about using Firebase with Angular applications:
 
-* Browse the [AngularFire API documentation](https://angularfire.firebaseapp.com/api.html).
+* Browse the [AngularFire API reference](/docs/reference.md).
```
Comment (katowulf) [inline docs/guide/beyond-angularfire.md]: Same here, think the leading `/` is incorrect?

File: docs/reference.md [hunk 3]
```diff
@@ -0,0 +1,1237 @@
...
+]);
+```
+
+``` html
+<!-- HTML -->
+<li ng-repeat="item in list | filter:name">{{ item | json }}</li>
+```
+
+The `$firebaseArray` service can also take a
+[query](https://firebase-dot-devsite.googleplex.com/docs/database/web/retrieve-data) to only sync
```
Comment (katowulf) [inline docs/reference.md]: I think this is a staging URL? Should be https://firebase.google.com/docs/database/web/retrieve-data?

File: docs/reference.md [hunk 4]
```diff
@@ -0,0 +1,1237 @@
...
+]);
+```
+
+The authentication object returned by `$firebaseAuth` contains several methods for authenticating
+users, responding to changes in authentication state, and managing user accounts for email /
+password users.
+
+### $signInWithCustomToken(authToken)
+
+Authenticates the client using a [custom authentication token](https://firebase-dot-devsite.googleplex.com/docs/auth/web/custom-auth).
```
Comment (katowulf) [inline docs/reference.md]: Another staging url.

File: docs/reference.md [hunk 5]
```diff
@@ -0,0 +1,1237 @@
...
+}).catch(function(error) {
+  console.error("Authentication failed:", error);
+});
+```
+
+This method returns a promise which is resolved or rejected when the authentication attempt is
+completed. If successful, the promise will be fulfilled with an object containing the payload of
+the authentication token. If unsuccessful, the promise will be rejected with an `Error` object.
+
+Read [our documentation on Custom Login](https://firebase-dot-devsite.googleplex.com/docs/auth/web/custom-auth)
```
Comment (katowulf) [inline docs/reference.md]: Staging url.

File: docs/reference.md [hunk 6]
```diff
@@ -0,0 +1,1237 @@
...
+}).catch(function(error) {
+  console.error("Authentication failed:", error);
+});
+```
+
+This method returns a promise which is resolved or rejected when the authentication attempt is
+completed. If successful, the promise will be fulfilled with an object containing authentication
+data about the logged-in user. If unsuccessful, the promise will be rejected with an `Error` object.
+
+Read [our documentation on anonymous authentication](https://firebase-dot-devsite.googleplex.com/docs/auth/web/anonymous-auth)
```
Comment (katowulf) [inline docs/reference.md]: url ❄

File: docs/reference.md [hunk 7]
```diff
@@ -0,0 +1,1237 @@
...
+}).catch(function(error) {
+  console.error("Authentication failed:", error);
+});
+```
+
+This method returns a promise which is resolved or rejected when the authentication attempt is
+completed. If successful, the promise will be fulfilled with an object containing authentication
+data about the logged-in user. If unsuccessful, the promise will be rejected with an `Error` object.
+
+Read [our documentation on email / password authentication](https://firebase-dot-devsite.googleplex.com/docs/auth/web/password-auth)
```
Comment (katowulf) [inline docs/reference.md]: url 🔥

**Comment by katowulf** [comment]
@abehaskins this almost lgtm! There are two items I'm concerned about:

1) I think https://firebase-dot-devsite.googleplex.com is a staging URL and needs to be replaced with f.g.c.
2) A lot of the readmes start with a leading `/` and I think they won't work when we commit?

File: docs/guide/beyond-angularfire.md (hunk 2 above)
Comment (abehaskins) [inline docs/guide/beyond-angularfire.md]: This is correct - we've doubled checked all these links, Github is smart enough to handle the repo-specific /

**Comment by abehaskins** [comment]
Fixed staging URLs - great catch

The `/`s work fine though, check out [this file](https://github.com/firebase/angularfire/blob/docs-ref/docs/guide/beyond-angularfire.md#next-steps) and try the reference link :)

File: README.md [hunk 8]
```diff
@@ -66,7 +66,7 @@ $ bower install angularfire --save
 
 * [Quickstart](docs/quickstart.md)
 * [Guide](docs/guide/README.md)
-* [API Reference](https://angularfire.firebaseapp.com/api.html)
+* [API Reference](docs/reference.md)
```
Comment (katowulf) [inline README.md]: > The /s work fine though, check out this file and try the reference link :)

In that case, is this one incorrect? Shouldn't it also begin with a /? Sigh, I feel like a GitHub noob suddenly. I didn't know I was so poorly versed in how URLs work.

**Comment by katowulf** [comment]
One last comment, Abe! Then we're ready to merge.

File: README.md (hunk 8 above)
Comment (jwngr) [inline README.md]: No, this page is at the root of the repo, so you don't need the leading `/`. It's just like doing `require()` in Node. Relative paths are relative to the current file and absolute paths are relative to the root of the repo.

**Comment by jwngr** [comment]
Looks good Abe!

**Pull request merged**